File: lib/rest/yampi/mappers/yampi_category_mapper.dart
import 'package:sertton/core/catalog/dtos/category_dto.dart';
import 'package:sertton/rest/types/json.dart';

class YampiCategoryMapper {
  static CategoryDto toDto(Json json) {
    return CategoryDto(
      id: json['id'].toString(),
      name: json['name'],
      description: json['description'],
    );
  }

  static List<CategoryDto> toDtoList(Json json) {
    final data = (json['data'] as List).cast<Json>();
    return data.map(toDto).toList();
  }
}

File: src/rest/yampi/mappers/yampi-category-mapper.ts
import type { Category } from "@/core/catalog/entities"
import type { YampiCategory } from "@/rest/yampi/types"

export const YampiCategoryMapper = () => ({
  toDomain(input: YampiCategory): Category {
    return { id: String(input.id), name: input.name, description: input.description ?? "" }
  },
})
